lib/hex_web/email.ex

```diff
@@ -1,11 +1,78 @@
 defmodule HexWeb.Email do
-  @type emails :: [String.t]
-  @type title  :: String.t
-  @type body   :: Phoenix.HTML.Safe
+  import Bamboo.Email
+  use Bamboo.Phoenix, view: HexWeb.EmailView
 
-  @callback send(emails, title, body) :: term
+  def owner_added(package, owners, owner) do
+    new_email
+    |> to(owners)
+    |> from(source)
+    |> subject("Hex.pm - Owner added")
+    |> assign(:username, owner.username)
+    |> assign(:email, owner.email)
+    |> assign(:package, package.name)
+    |> render("owner_add.html")
+  end
 
-  @email_impl Application.get_env(:hex_web, :email_impl)
+  def owner_removed(package, owners, owner) do
+    new_email
+    |> to(owners)
+    |> from(source)
+    |> subject("Hex.pm - Owner removed")
+    |> assign(:username, owner.username)
+    |> assign(:email, owner.email)
+    |> assign(:package, package.name)
+    |> render("owner_remove.html")
+  end
 
-  defdelegate send(emails, title, body), to: @email_impl
+  def user_confirmed(user) do
+    new_email
+    |> to(user)
+    |> from(source)
+    |> subject("Hex.pm - Account confirmed")
+    |> render("confirmed.html")
+  end
+
+  def confirmation_request(user) do
+    new_email
+    |> to(user)
+    |> from(source)
+    |> subject("Hex.pm - Account confirmation")
+    |> assign(:username, user.username)
+    |> assign(:key, user.confirmation_key)
+    |> render("confirmation_request.html")
+  end
+
+  def password_reset_request(user) do
+    new_email
+    |> to(user)
+    |> from(source)
+    |> subject("Hex.pm - Password reset request")
+    |> assign(:username, user.username)
+    |> assign(:key, user.reset_key)
+    |> render("password_reset_request.html")
+  end
+
+  def password_reset(user) do
+    new_email
+    |> to(user)
+    |> from(source)
+    |> subject("Hex.pm - Password reset")
+    |> render("password_reset.html")
+  end
+
+  def typosquat_candidates([], _), do: :ok
+  def typosquat_candidate(candidates, threshold) do
+    new_email
+    |> to(Application.get_env(:hex_web, :support_email))
+    |> from(source)
+    |> subject("Hex.pm - Typosquat candidates")
+    |> assign(:candidates, candidates)
+    |> assign(:threshold, threshold)
+    |> render("typosquat_candidates.html")
+  end
+
+  defp source do
+    source = Application.get_env(:hex_web, :email_host) || "noreply@hex.pm"
```

Comment:
I think this evaluates to: `"hex.pm" || "noreply@hex.pm"` which would be incorrect. Previously code was: `"noreply@" <> source`